lua/telescope/actions/init.lua

```diff
@@ -1020,7 +1020,9 @@ end
 actions.delete_buffer = function(prompt_bufnr)
   local current_picker = action_state.get_current_picker(prompt_bufnr)
   current_picker:delete_selection(function(selection)
-    vim.api.nvim_buf_delete(selection.bufnr, { force = false })
+    local force = vim.api.nvim_buf_get_option(selection.bufnr, "buftype") == "terminal"
+    local ok = pcall(vim.api.nvim_buf_delete, selection.bufnr, { force = force })
```

Comment:
In my testing, wrapping `nvim_buf_delete` with a `pcall` will not catch the vim error caused by trying to delete an unsaved/changed buffer.

Maybe we can add something like below above this line so users don't see a big error
```lua
if vim.fn.getbufinfo(selection.bufnr)[1].changed then
  -- maybe add in a less intrusive msg
  return
end
```